cobra/flux_analysis/sampling.py

```diff
@@ -113,55 +186,60 @@ def generate_fva_warmup(self, solver=None, **solver_args):
-                # Omit fixed reactions
-                if self.fixed[i]:
-                    pass
-                solver.change_variable_objective(lp, i, 1.0)
-                solver.solve_problem(lp, objective_sense=sense, **solver_args)
-                sol = solver.format_solution(lp, self.model).x
-                if not sol:
-                    pass
-                # some solvers do not enforce bounds too much -> we reconstrain
-                sol = np.maximum(sol, self.bounds[0, ])
-                sol = np.minimum(sol, self.bounds[1, ])
-                self.warmup[self.n_warmup, ] = sol
-                self.n_warmup += 1
-                # revert objective
-                solver.change_variable_objective(lp, i, 0.)
+        idx = np.hstack([self.fwd_idx, self.rev_idx])
+        self.warmup = np.zeros((len(idx), len(self.model.variables)))
+        self.model.objective = S.Zero
+        self.model.objective.direction = "max"
+        variables = self.model.variables
+        for i in idx:
+            # Omit fixed reactions
+            if self.problem.fixed[i]:
+                pass
+            self.model.objective.set_linear_coefficients({variables[i]: 1})
+            self.model.solver.optimize()
+            try:
+                assert_optimal(self.model)
+            except Exception:
+                continue
+            primals = self.model.solver.primal_values
+            sol = [primals[v.name] for v in self.model.variables]
+            # some solvers do not enforce bounds too much -> we reconstrain
+            sol = np.maximum(sol, self.problem.var_bounds[0, ])
+            sol = np.minimum(sol, self.problem.var_bounds[1, ])
+            self.warmup[self.n_warmup, ] = sol
+            self.n_warmup += 1
+            # revert objective
+            self.model.objective.set_linear_coefficients({variables[i]: 0})
         # Shrink warmup points to measure
         self.warmup = self.warmup[0:self.n_warmup, ]
 
     def _reproject(self, p):
-        """Reprojects a point into the feasibility region"""
-        return self.NS.dot(self.NS.T.dot(p))
+        """Reproject a point into the feasibility region."""
+        return self.problem.projection.dot(p)
 
     def _bounds_dist(self, p):
         """Get the lower and upper bound distances. Negative is bad."""
-        lb_dist = (p - self.bounds[0, ]).min()
-        ub_dist = (self.bounds[1, ] - p).min()
+        prob = self.problem
+        lb_dist = (p - prob.var_bounds[0, ]).min()
+        ub_dist = (prob.var_bounds[1, ] - p).min()
+        if prob.bounds.shape[0] > 0:
+            const = prob.inequalities.dot(p)
+            const_lb_dist = (const - prob.bounds[0, ]).min()
+            const_ub_dist = (prob.bounds[1, ] - const).min()
+            lb_dist = min(lb_dist, const_lb_dist)
+            ub_dist = min(ub_dist, const_ub_dist)
         return np.array([lb_dist, ub_dist])
 
-    def sample(self, n):
+    def sample(self, n, fluxes=True):
```

Comment:
In theory yes, only problem is that if you change it continuously you might not know anymore with which thinning earlier samplers were generated. I right now it is also possible to do that, just reassign sampler.thinning (it is not protected).